cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -331,7 +323,13 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration) ext.Deployment
 			},
 		},
 	}
-	return d
+}
+
+func prependSelfHostedFlock(name string, args []string) []string {
+	// TODO: flock has to be inside the container, and a shell must exist with this approach
+	// Since the flock binary is present at /bin/flock in a busybox image and /usr/bin/flock in a debian image, we can't hardcode the path
```

Comment:
Again, flock is part of all Linux distros that I'm aware of and if the idea is to remove the most basic commands from these containers, then go with unikernels + the component (e.g. apiserver) static binary.